CCDB/include/CCDB/CcdbApi.h

```diff
@@ -0,0 +1,175 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   CcdbApi.h
+/// \author Barthelemy von Haller
+///
+
+#ifndef PROJECT_CCDBAPI_H
+#define PROJECT_CCDBAPI_H
+
+#include <string>
+#include <memory>
+#include <iostream>
+#include <map>
+#include <curl/curl.h>
+#include <TObject.h>
+
+namespace o2
+{
+namespace ccdb
+{
+
+/**
+ * Interface to the CCDB.
+ * It uses Curl to talk to the REST api.
+ *
+ * @todo use smart pointers ?
+ * @todo handle errors and exceptions
+ * @todo extend code coverage
+ */
+class CcdbApi //: public DatabaseInterface
+{
+ public:
+  /// \brief Default constructor
+  CcdbApi();
+  /// \brief Default destructor
+  virtual ~CcdbApi();
+
+  /**
+   * Initialize connection to CCDB
+   *
+   * @param host The URL to the CCDB (e.g. "ccdb-test.cern.ch:8080")
+   */
+  void init(std::string host);
+
+  /**
+   * Stores an object in the CCDB
+   *
+   * @param rootObject Shared pointer to the object to store.
+   * @param path The path where the object is going to be stored.
+   * @param metadata Key-values representing the metadata for this object.
+   * @param startValidityTimestamp Start of validity. If omitted, current timestamp is used.
+   * @param endValidityTimestamp End of validity. If omitted, current timestamp + 1 year is used.
+   */
+  void store(TObject* rootObject, std::string path, std::map<std::string, std::string> metadata,
```

Comment:
I suggest to return the newly created object ID / path from the store() method

Reply:
I agree on the idea but fail to implement it. I tried to use 
```
    std::string requestOutput;
    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, CurlWrite_CallbackFunc_StdString);
    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &requestOutput);
```
It works in the method list() but here it does not. Probably because of the multipart or because of the redirection. 

I propose to keep it as a future improvement if you don't mind.